services/worker-manager/src/provisioner.js

```diff
@@ -142,20 +142,12 @@ class Provisioner {
       };
 
       // Check the state of workers (state is updated by worker-scanner)
-      await this.Worker.scan({}, {
-        handler: (worker) => {
-          // We only support conditions on dates, as they cannot
-          // be used to inject SQL -- `Date.toJSON` always produces a simple string
-          // with no SQL metacharacters.
-          //
-          // Previously with azure, we added the query in the scan method
-          // (i.e., this.Worker.scan({ state: ... })) but since the query doesn't include
-          // the partition key or row key, we would need to manually filter through
-          // the table.
-          if (worker.state !== this.Worker.states.STOPPED) {
-            seen(worker);
-          }
-        },
+      const workers = await Worker.getWorkers(this.db, {});
```

Comment:
We'll want to add another method (e.g., get_workers_non_running) since get_workers only takes `=` as operator. Do you want me to do that?

Reply:
The `get_workers` db function takes "worker_pool_id_in text, worker_group_in text, worker_id_in text, state_in text, page_size_in integer, page_offset_in integer" and filters by state.  So I think this would just be updating `Worker.getWorkers` to allow passing that argument, right?

Reply:
Right. I'll see what I can do.

Reply:
I was thinking about this for my `existing_capacity` change and was thinking we should actually have a db function that just counts workers in a state. This can be as simple as a `select count(*)` for us since these tables don't really get that big. `provisioner.js` really only gets these workers to count how many are in a state anyway and doesn't do anything with the workers themselves. This function could then also be used to return this data in the api without, in a sense, denormalizing the data about how many workers we have.

I think I built most of my existing_capacity PR still in the mindset of azure tables but we have the power of sql now. Does it make sense to use it here? In general it feels like getting a lot of this vaguely complicated logic into the db itself is good.